File: fix_login.py
import re

with open("login.html", "r", encoding="utf-8") as f:
    content = f.read()

# We need to wrap <span class="input-prefix">... <input ...> and optional <button class="pw-toggle">
# inside <div style="position: relative;">
def replacer(match):
    inner = match.group(1)
    # The inner content usually starts with <span class="input-prefix">
    # We want to wrap everything up to the <input> or <button>
    # Let's just find the <div class="field-feedback" or <div class="pw-strength"
    # and put everything before it into a wrapper.
    parts = re.split(r'(<div class="(?:pw-strength|field-feedback)")', inner, 1)
    if len(parts) > 1:
        wrapped = f'\n        <div style="position: relative;">{parts[0].rstrip()}\n        </div>\n        {parts[1]}{parts[2]}'
        return f'<div class="auth-input-group">{wrapped}</div>'
    return match.group(0)

new_content = re.sub(r'<div class="auth-input-group">(.*?)</div>\s*(?=</div|<div|<button)', replacer, content, flags=re.DOTALL)

with open("login.html", "w", encoding="utf-8") as f:
    f.write(new_content)
print("Done")

File: fix_login.js
const fs = require("fs");
let content = fs.readFileSync("login.html", "utf-8");

// We want to find the blocks starting with <div class="auth-input-group"> and ending with </div>
// But they contain nested divs!
// Let's use a simpler approach: 
// 1. Split by '<div class="auth-input-group">'
// 2. For each part, find where `<div class="field-feedback"` or `<div class="pw-strength"` starts.
// 3. Insert `<div style="position: relative;">` at the beginning, and `</div>` right before the feedback/strength div.

let parts = content.split('<div class="auth-input-group">');
let newContent = parts[0];

for (let i = 1; i < parts.length; i++) {
    let part = parts[i];
    // Find the split point
    let splitIdx = part.indexOf('<div class="field-feedback"');
    let strengthIdx = part.indexOf('<div class="pw-strength"');
    
    if (strengthIdx !== -1 && strengthIdx < splitIdx) {
        splitIdx = strengthIdx;
    }
    
    if (splitIdx !== -1) {
        let before = part.substring(0, splitIdx);
        let after = part.substring(splitIdx);
        
        // Wrap 'before'
        let wrapped = `\n        <div style="position: relative;">` + before.trimEnd() + `\n        </div>\n        `;
        newContent += '<div class="auth-input-group">' + wrapped + after;
    } else {
        newContent += '<div class="auth-input-group">' + part;
    }
}

// Now revert the CSS top: 24px to top: 50%
newContent = newContent.replace(/top: 24px;/g, "top: 50%;");

fs.writeFileSync("login.html", newContent, "utf-8");
console.log("Replaced successfully!");
